modules/virt-checking-cluster-dpdk-readiness.adoc

```diff
@@ -0,0 +1,212 @@
+
+Use a predefined checkup to verify if your {product-title} cluster node can run a virtual machine (VM) with a Data Plane Development Kit (DPDK) workload. The checkup runs traffic between a traffic generator pod and a VM running a test DPDK application, and checks for packet loss.
+
+.Prerequisites
+* You have access to the cluster as a user with `cluster-admin` permissions.
+* You have installed the OpenShift CLI (`oc`).
+* You have configured the compute nodes to run DPDK applications on VMs with zero packet loss.
+
+.Procedure
+. Create a manifest file that contains the `ServiceAccount`, `Role`, and `RoleBinding` objects with permissions that the checkup requires for cluster access:
++
+.Example roles manifest
+[%collapsible]
+====
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: dpdk-checkup-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kiagnose-configmap-access
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "configmaps" ]
+    verbs: [ "get", "update" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kiagnose-configmap-access
+subjects:
+  - kind: ServiceAccount
+    name: dpdk-checkup-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kiagnose-configmap-access
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubevirt-dpdk-checker
+rules:
+  - apiGroups: [ "kubevirt.io" ]
+    resources: [ "virtualmachineinstances" ]
+    verbs: [ "create", "get", "delete" ]
+  - apiGroups: [ "subresources.kubevirt.io" ]
+    resources: [ "virtualmachineinstances/console" ]
+    verbs: [ "get" ]
+  - apiGroups: [ "" ]
+    resources: [ "pods" ]
+    verbs: [ "create", "get", "delete" ]
+  - apiGroups: [ "" ]
+    resources: [ "pods/exec" ]
+    verbs: [ "create" ]
+  - apiGroups: [ "k8s.cni.cncf.io" ]
+    resources: [ "network-attachment-definitions" ]
+    verbs: [ "get" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kubevirt-dpdk-checker
+subjects:
+  - kind: ServiceAccount
+    name: dpdk-checkup-sa
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kubevirt-dpdk-checker
+----
+====
+
+. Apply the checkup roles manifest:
++
+[source,terminal]
+----
+$ oc apply -n <target_namespace> -f <dpdk_roles>.yaml
+----
+
+. Create a `ConfigMap` manifest that contains the input parameters for the checkup. The config map also stores the results of the checkup.
++
+.Example input config map
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dpdk-checkup-config
+data:
+  spec.timeout: 10m
+  spec.param.networkAttachmentDefinitionName: <network_name> <1>
+  spec.param.trafficGeneratorRuntimeClassName: <runtimeclass_name> <2>
+----
+<1> The name of the `NetworkAttachmentDefinition` object.
+<2> The `RuntimeClass` resource that the traffic generator pod uses.
+
```

Comment:
I think that a reference to a table (e.g. https://github.com/kiagnose/kubevirt-dpdk-checkup#configuration, assuming  that referencing an upstream repo from a downstream documentation is valid) of the parameters that can be configured using the ConfigMap can be useful, otherwise how can the user know what parameters are available for them to configure?

Reply:
Yossi, Linking to GitHub is not normally permitted from downstream docs. It requires permission.

Shikha, I ran the table through kramdown-asciidoc to convert the format. I will send you the file via Slack and you can decide whether you want to use it or not.

Reply:
Thanks, Avital. @yossisegev I will add the configuration parameters table and update the PR soon.